ch_backup/storage/engine/s3/s3_engine.py

```diff
@@ -88,15 +89,36 @@ def delete_files(self, remote_paths: Sequence[str]) -> None:
         Delete multiple files from S3
         """
         # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.delete_objects
-        if self._bulk_delete_enabled:
-            objects_to_delete: Sequence = [
+        if not self._bulk_delete_enabled:
+            for remote_path in remote_paths:
+                self.delete_file(remote_path)
+            return
+
+        try:
+            objects_to_delete: list = [
                 {"Key": path.lstrip("/")} for path in remote_paths
             ]
             self._s3_client.delete_objects(
                 Bucket=self._s3_bucket_name, Delete={"Objects": objects_to_delete}
             )
-        else:
-            for remote_path in remote_paths:
+        except ClientError as e:
+            if "MalformedXML" not in repr(e):
+                raise
+
+            objects_to_delete_single: list = []
+            objects_to_delete_bulk: list = []
+            for path in remote_paths:
+                if contains_unusual_characters(path):
+                    objects_to_delete_single.append(path)
+                else:
+                    objects_to_delete_bulk.append({"Key": path.lstrip("/")})
+
+            if objects_to_delete_bulk:
+                self._s3_client.delete_objects(
+                    Bucket=self._s3_bucket_name,
+                    Delete={"Objects": objects_to_delete_bulk},
+                )
+            for remote_path in objects_to_delete_single:
                 self.delete_file(remote_path)
```

Comment:
I think fallback logic can be simplified. If bulk delete fails with `MalformedXML` then simply delete all files one by one.